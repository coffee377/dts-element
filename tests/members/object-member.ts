import * as dts from '../index';

it('should return correctly with owned (VariableDeclaration)', () => {
  expect(dts.emit(
    dts.create_object_member({
      owned: dts.create_variable_declaration({
        name: 'a',
      }),
    }),
  )).toMatchSnapshot();
});

it('should return correctly with owned (VariableDeclaration), optional, readonly', () => {
  expect(dts.emit(
    dts.create_object_member({
      owned: dts.create_variable_declaration({
        name: 'a',
      }),
      optional: true,
      readonly: true,
    }),
  )).toMatchSnapshot();
});

it('should return correctly with owned (FunctionDeclaration with undefined name)', () => {
  expect(dts.emit(
    dts.create_object_member({
      owned: dts.create_function_declaration({
        name: undefined,
      }),
    }),
  )).toMatchSnapshot();
});

it('should return correctly with owned (FunctionDeclaration with name)', () => {
  expect(dts.emit(
    dts.create_object_member({
      owned: dts.create_function_declaration({
        name: 'fn',
      }),
    }),
  )).toMatchSnapshot();
});

it('should return correctly with owned (FunctionDeclaration), optional', () => {
  expect(dts.emit(
    dts.create_object_member({
      owned: dts.create_function_declaration({
        name: 'fn',
      }),
      optional: true,
    }),
  )).toMatchSnapshot();
});

it('should return correctly with owned (ConstructorType)', () => {
  expect(dts.emit(
    dts.create_object_member({
      owned: dts.create_constructor_type(),
    }),
  )).toMatchSnapshot();
});

it('should throw error with unexpected kind', () => {
  expect(() => dts.emit(
    dts.create_object_member({
      owned: dts.any_type as any,
    }),
  )).toThrowError();
});
