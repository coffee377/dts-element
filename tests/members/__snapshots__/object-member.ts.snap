// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correctly with owned (FunctionDeclaration with name) 1`] = `"fn(): any;"`;

exports[`should return correctly with owned (FunctionDeclaration with undefined name) 1`] = `"(): any;"`;

exports[`should return correctly with owned (FunctionDeclaration), optional 1`] = `"fn?(): any;"`;

exports[`should return correctly with owned (VariableDeclaration) 1`] = `"a: any;"`;

exports[`should return correctly with owned (VariableDeclaration), optional, readonly 1`] = `"readonly a?: any;"`;
